docs/user/openstack/troubleshooting.md

```diff
@@ -0,0 +1,79 @@
+# OpenShift 4 installer on OpenStack troubleshooting
+
+Support for launching clusters on OpenStack is **experimental**.
+
+Unfortunately, there will always be some cases where OpenShift fails to install properly. In these events, it is helpful to understand the likely failure modes as well as how to troubleshoot the failure.
+
+This document discusses some troubleshooting options for OpenStack based
+deployments. For general tips on troubleshooting the installer, see the [Installer Troubleshooting](../troubleshooting.md) guide.
+
+## View instances logs
+
+OpenStack CLI tools should be installed, then:
+
+`openstack console log show <instance>`
+
+## ssh access to the instances
+
+By default, the only exposed instance is the service VM, but ssh access is not
+allowed (nor the ssh key injected), so in case you want to access the hosts, it
+is required to create a floating IP and attach it to some host (master-0 in
+this example)
+
+### Create security group to allow ssh access
+
+```
+INSTANCE=$(openstack server list -f value -c Name | grep master-0)
+openstack security group create ssh
+# Note this opens port 22/tcp to 0.0.0.0/0
+openstack security group rule create \
+  --ingress \
+  --protocol tcp \
+  --dst-port 22 \
+  ssh
+openstack server add security group ${INSTANCE} ssh
+```
+
+Optionally, allow ICMP traffic (to ping the instance):
+
+```
+openstack security group rule create \
+  --ingress \
+  --protocol icmp \
+  ssh
+```
+
+### Create and attach the floating IP
+
+```
+# This assumes there are only two networks, the openshift one and the public one
+PUBLIC_NETWORK=$(openstack network list -f value -c Name | grep -v openshift)
```

Comment:
Everything else can by copy/pasted, but this will likely fail (as you point out in the comment).  How about just a note saying that PUBLIC_NETWORK should be set to the same public network as configured in the openshift install?

```
# NOTE: This must be updated to match the external network configured in the openshift install.
PUBLIC_NETWORK=public
```

or something like that?